lightning-background-processor/src/lib.rs

```diff
@@ -173,7 +173,7 @@ impl BackgroundProcessor {
 		Descriptor: 'static + SocketDescriptor + Send + Sync,
 		CMH: 'static + Deref + Send + Sync,
 		RMH: 'static + Deref + Send + Sync,
-		EH: 'static + EventHandler + Send + Sync,
+		EH: 'static + EventHandler + Send,
```

Comment:
Do we want to make it explicit that the user pass an `Option<Deref<InvoidPayer>>` to the background processor, which does the decoration in `start`, vs the user doing the decoration before calling start?

Reply:
I think the decorator approach may be required or at least preferred in this case. The problem with passing `InvoicePayer` to `BackgroundProcessor` is that the conditional logic for when to call the user's handler is encapsulated in `InvoicePayer`. We'd need to leak that information somehow to the `DecoratingEventHandler` so it knows when to call the user's handler (i.e. either when there are no more retries or for events other than `PaymentFailed`).

Reply:
Ah, yea, I was somehow thinking you'd have the BP update the inner handler that the InvoicePayer holds, which is maybe not practical/clean. Its probably not entirely crazy to have a simpler wrapper struct/eventhandler in BP that calls back into the user-provided event handler - do you think that's practical? I guess the only other alternative is to have the BP construct the InvoicePayer internally and then expose it, which is maybe veering too far into the "BP is a node" direction.

Reply:
Given both `BackgroundProcessor` and `InvoicePayer` consume the user's event handler, I think the decorator approach is probably ok here. They'd have to go out of their way to get it wrong. And we could make sure `InvoicePayer`'s docs make it clear about how to use it.